File: packages/dart/lib/src/word/rand_tech.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// The vocabulary of computers and the networks between them.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.tech] to `randWordDetails` — Dart has
/// no overloads, and twenty-four more functions for it would be twenty-four too many.
///
/// ```dart
/// randTech(language: WordLanguage.ko, count: 3); // [서버, 캐시, 대역]
/// randTech(language: WordLanguage.en, count: 3); // [Server, Cache, Subnet]
/// ```
List<String> randTech({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.tech,
  count: count,
  realism: realism,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);

File: packages/javascript/lib/word/randTech.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * The vocabulary of computers and the networks between them.
 *
 * `randWord({ theme: 'tech' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randTech({ language: 'ko', count: 3 }); // ['서버', '캐시', '대역']
 * randTech({ language: 'en', count: 3 }); // ['Server', 'Cache', 'Subnet']
 */
export function randTech(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Tech words with the language and theme each one came from.
 *
 * @example
 * randTech({ language: 'ko', output: 'detail' });
 * // [{ word: '서버', language: 'ko', theme: 'tech' }]
 */
export function randTech(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randTech(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('tech', options);
}
